java/vector/src/main/java/org/apache/arrow/vector/VectorLoader.java

```diff
@@ -98,7 +98,7 @@ private void loadBuffers(
       CompressionCodec codec) {
     checkArgument(nodes.hasNext(), "no more field nodes for field %s and vector %s", field, vector);
     ArrowFieldNode fieldNode = nodes.next();
-    int bufferLayoutCount = TypeLayout.getTypeBufferCount(field.getType());
+    int bufferLayoutCount = TypeLayout.getTypeBufferCount(field.getType(), vector);
```

Comment:
Does this make sense? The vector is possibly uninitialized right now. You need to use the IPC metadata to figure out the buffer count here, not the Java vector.

Reply:
Sorry, @lidavidm , I have a knowledge gap here. In this particular method we have access to the iterator with buffers. Would it be ideal if we iterate until null? I know this gives a non-initialized list. Is there an alternative?

Reply:
That means we're just blindly trusting the input. 

I recommend reviewing the spec again... https://github.com/apache/arrow/blob/main/format/Message.fbs#L117

Reply:
Oh I see. So it seems we should update the `RecordBatch` to include `variadicBufferCounts` then we can use that information in such cases where the vector has not been initialized. That's has not yet been integrated. So probably that needs to be done here as well.

Reply:
@lidavidm When we add this attribute to the `RecordBatch` message, should we populate non View-type vectors with `0` values or just leave it as it is but customize to pass `variadicBufferCounts` for View-type vectors?

Reply:
What would you put besides 0?

Reply:
I think it needs to be part of RecordBatch

Reply:
Yes of course. I added it and I am getting some memory leaks from non variable width vectors. Need to check it.that's where I was adding the non zero variadicBufferCount views and zero for others.

Reply:
That's why I was thinking whether to not populate that field for non-views.